dd-trace-propagation/src/datadog.rs

```diff
@@ -123,35 +123,33 @@ fn get_propagation_tags(
     tags: &HashMap<String, String>,
     max_length: usize,
 ) -> Result<String, Error> {
-    // Use a single String buffer to avoid intermediate Vec allocation
-    let mut propagation_tags = String::new();
-    let mut first = true;
-
-    for (key, value) in tags.iter() {
-        if !key.starts_with(DATADOG_PROPAGATION_TAG_PREFIX) {
-            continue;
-        }
+    // Compute size before writing to prevent reallocations
+    let total_size: usize = tags
+        .iter()
+        .filter(|(k, _)| k.starts_with(DATADOG_PROPAGATION_TAG_PREFIX))
+        .enumerate()
+        .map(|(i, (k, v))| k.len() + v.len() + 1 + if i == 0 { 0 } else { 1 })
```

Comment:
nit: A comment explaining why `+ 1 + if i == 0 { 0 } else { 1 }` is necessary would be helpful to the next person who has to read this.